x-pack/test/plugin_functional/plugins/resolver_test/public/applications/resolver_test/index.tsx

```diff
@@ -4,119 +4,71 @@
  */
-export function renderApp(
-  { element }: AppMountParameters,
-  embeddable: Promise<IEmbeddable | undefined>
-) {
+export function renderApp(context: AppMountContext, parameters: AppMountParameters) {
   /**
    * The application DOM node should take all available space.
    */
-  element.style.display = 'flex';
-  element.style.flexGrow = '1';
+  parameters.element.style.display = 'flex';
+  parameters.element.style.flexGrow = '1';
 
-  ReactDOM.render(
-    <I18nProvider>
-      <AppRoot embeddable={embeddable} />
-    </I18nProvider>,
-    element
-  );
+  ReactDOM.render(<AppRoot context={context} parameters={parameters} />, parameters.element);
 
   return () => {
-    ReactDOM.unmountComponentAtNode(element);
+    ReactDOM.unmountComponentAtNode(parameters.element);
   };
 }
 
-const AppRoot = styled(
-  React.memo(
-    ({
-      embeddable: embeddablePromise,
-      className,
-    }: {
-      /**
-       * A promise which resolves to the Resolver embeddable.
-       */
-      embeddable: Promise<IEmbeddable | undefined>;
-      /**
-       * A `className` string provided by `styled`
-       */
-      className?: string;
-    }) => {
-      /**
-       * This state holds the reference to the embeddable, once resolved.
-       */
-      const [embeddable, setEmbeddable] = React.useState<IEmbeddable | undefined>(undefined);
-      /**
-       * This state holds the reference to the DOM node that will contain the embeddable.
-       */
-      const [renderTarget, setRenderTarget] = React.useState<HTMLDivElement | null>(null);
-
-      /**
-       * Keep component state with the Resolver embeddable.
-       *
-       * If the reference to the embeddablePromise changes, we ignore the stale promise.
-       */
-      useEffect(() => {
-        /**
-         * A promise rejection function that will prevent a stale embeddable promise from being resolved
-         * as the current eembeddable.
-         *
-         * If the embeddablePromise itself changes before the old one is resolved, we cancel and restart this effect.
-         */
-        let cleanUp;
-
-        const cleanupPromise = new Promise<never>((_resolve, reject) => {
-          cleanUp = reject;
-        });
-
-        /**
-         * Either set the embeddable in state, or cancel and restart this process.
-         */
-        Promise.race([cleanupPromise, embeddablePromise]).then((value) => {
-          setEmbeddable(value);
-        });
+const AppRoot = React.memo(
+  ({ context, parameters }: { context: AppMountContext; parameters: AppMountParameters }) => {
+    const dataAccessLayer: DataAccessLayer = useMemo(
+      () => resolverMockDataAccessLayerWithNoAncestorsTwoChildren().dataAccessLayer,
+      []
+    );
 
-        /**
-         * If `embeddablePromise` is changed, the cleanup function is run.
-         */
-        return cleanUp;
-      }, [embeddablePromise]);
+    const store = useMemo(() => {
+      return resolverStoreFactory(dataAccessLayer);
```

Comment:
create a store for resolver using the mock data access layer (so that it communicates with the mock instead of a server.)